From eba39eaae53640d87e72d6a5ea7593044ccf2d6b Mon Sep 17 00:00:00 2001
From: Anish Trivedi <anish@freescale.com>
Date: Mon, 18 Apr 2011 15:54:19 -0500
Subject: [PATCH 218/505] ENGR00142299 [imx-test] SRTC: Change interrupts
 expected for 2.6.38 kernel

RTC driver in new kernel uses SRTC hw interrupts for the
update interrupts (UIE), therefore change expected interrupts
to 11 instead of 1 for the new kernel.

Signed-off-by: Anish Trivedi <anish@freescale.com>
---
 test/mxc_rtc/autorun-rtc.sh | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/test/mxc_rtc/autorun-rtc.sh b/test/mxc_rtc/autorun-rtc.sh
index f287a61..aff1b39 100755
--- a/test/mxc_rtc/autorun-rtc.sh
+++ b/test/mxc_rtc/autorun-rtc.sh
@@ -11,7 +11,14 @@ echo "$(platform)"
 if [ "$(platform)" == IMX37_3STACK ] || [ "$(platform)" == IMX51 ] \
     || [ "$(platform)" == IMX53 ] || [ "$(platform)" == IMX50 ]; then
 	rtc_test_param=--no-periodic
-	RTC_IRQS_EXPECTED=1
+	# For kernel version 2.6.38 and higher, number of interrupts expected will be 11
+	VER=`echo $(kernel_version) | cut -d. -f3`
+	REF_VER=38
+	if [ $VER -ge $REF_VER ]; then
+		RTC_IRQS_EXPECTED=11
+	else
+		RTC_IRQS_EXPECTED=1
+	fi
 else
 	rtc_test_param=--full
 	RTC_IRQS_EXPECTED=131
-- 
1.8.0

